verilog/CST/type.h

```diff
@@ -61,6 +61,39 @@ bool IsStorageTypeOfDataTypeSpecified(const verible::Symbol&);
 const verible::SyntaxTreeLeaf* GetIdentifierFromTypeDeclaration(
     const verible::Symbol& symbol);
 
+// Extracts kReferenceCallBase node from nodes tagged with kInstantiationType.
+const verible::SyntaxTreeNode& GetReferenceCallBaseFromInstantiationType(
+    const verible::Symbol& instantiation_type);
+
+// Extracts kReference node from nodes tagged with kReferenceCallBase.
+const verible::SyntaxTreeNode& GetReferenceFromReferenceCallBase(
+    const verible::Symbol& reference_call_base);
+
+// Extracts kLocalRoot node from nodes tagged with kReference.
+const verible::SyntaxTreeNode& GetLocalRootFromReference(
+    const verible::Symbol& reference);
+
+// Extracts kUnqualifiedId node from nodes tagged with kLocalRoot.
+const verible::SyntaxTreeNode& GetUnqualifiedIdFromLocalRoot(
+    const verible::Symbol& local_root);
+
+// Extracts kUnqualifiedId node from nodes tagged with kReferenceCallBase.
+const verible::SyntaxTreeNode& GetUnqualifiedIdFromReferenceCallBase(
+    const verible::Symbol& reference_call_base);
+
+// Extracts kPackedDimensions node from node tagged with kDataTypePrimitive.
+const verible::SyntaxTreeNode& GetPackedDimensionFromDataType(
```

Comment:
Suggest copying/moving tests to type_test.cc to test closer to the module where defined.

Reply:
Use a FindAll function to get to the inner type nodes quickly without depending on descending through the full declaration.